File: src/main/java/robinsonade/Quantity.java
package robinsonade;

import java.util.Objects;

class Quantity {

    private final double value;

    static Quantity valueOf(double value) { return new Quantity(value); }

    private Quantity(double value) { this.value = value; }

    double multiplyBy(double value) { return this.value * value; }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Quantity quantity = (Quantity) o;
        return Double.compare(quantity.value, value) == 0;
    }

    @Override
    public int hashCode() {
        return Objects.hash(value);
    }

    @Override
    public String toString() {
        return "Quantity{" + "value=" + value + '}';
    }
}

File: CSharp/Robinsonade/Quantity.cs
namespace Robinsonade
{
    public class Quantity
    {
        private readonly double value;

        public static Quantity ValueOf(double value) { return new Quantity(value); }

        private Quantity(double value) { this.value = value; }

        public double MultiplyBy(double value) { return this.value * value; }

        public override bool Equals(object obj)
        {
            if (this == obj) return true;
            if (obj == null || this.GetType() != obj.GetType()) return false;
            Quantity quantity = obj as Quantity;
            return quantity.value.Equals(value);
        }

        public override string ToString()
        {
            return "Quantity{" + "value=" + value + '}';
        }
    }
}
